### PR TITLE
util/types: fix CoerceDatum function behavior

### PR DESCRIPTION
**Pull request by coocood**
When coerce two datums, float64 is a bigger container than decimal, so the result of compute decimal and float64 is float64.

### CODE REVIEW

**Comment by coocood** [comment]
@BusyJay @shenli @zimulala PTAL

File: util/types/datum.go [hunk 1]
```diff
@@ -1379,21 +1382,18 @@ func (d *Datum) convergeType(hasDecimal, hasFloat *bool) (x Datum) {
 // CoerceDatum changes type.
 // If a or b is Decimal, changes the both to Decimal.
 // Else if a or b is Float, changes the both to Float.
```
Comment (shenli) [inline util/types/datum.go]: Should we update the comment?

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

File: util/types/datum.go [hunk 2]
```diff
@@ -1407,7 +1408,12 @@ func CoerceDatum(a, b Datum) (x, y Datum) {
 			x.SetFloat64(x.GetMysqlEnum().ToNumber())
 		case KindMysqlSet:
 			x.SetFloat64(x.GetMysqlSet().ToNumber())
-
+		case KindMysqlDecimal:
+			fval, err := x.ToFloat64()
```
Comment (BusyJay) [inline util/types/datum.go]: Can fval be Inf?

Comment (coocood) [inline util/types/datum.go]: No, Decimal is a smaller container.

**Pull request merged**